### PR TITLE
feat(app-vite&app-webpack): Add PNPM support

### PR DESCRIPTION
**Pull request by GaliMu**
**What kind of change does this PR introduce?** 

- [ ] Bugfix
- [x] Feature
- [ ] Documentation
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch (or `v[X]` branch)
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar/tree/dev/docs)  or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to [start a new feature discussion](https://github.com/quasarframework/quasar/discussions/new?category=ideas-proposals) first and wait for approval before working on it)

**Other information:**
PNPM can only be correctly identified and used when executing a command, and it is uncertain whether the command can be executed correctly.

### CODE REVIEW

**Comment by rstoenescu** [comment]
Hi,

Thank you for contributing!
However, simply acknowledging pnpm as an option won't make it actually work. We do not officially support it (at least yet) due to its restrictions.

**Comment by yusufkandemir** [comment]
Re-opening as we have a bunch of PRs which will make it possible to use PNPM for both app-webpack and app-vite. See #13615 for a list of them.

File: app-vite/lib/helpers/node-packager.js [hunk 1]
```diff
@@ -139,11 +161,16 @@ function getPackager () {
...
+    return pnpm
+  }
+
   if (npm.isInstalled()) {
     return npm
   }
 
-  warn('Please install Yarn or NPM before running this command.\n')
+
+  warn('Please install Yarn or PNPM or NPM before running this command.\n')
```
Comment (yusufkandemir) [inline app-vite/lib/helpers/node-packager.js]:
```suggestion
  warn('Please install Yarn, PNPM, or NPM before running this command.\n')
```

File: app-webpack/lib/helpers/node-packager.js [hunk 2]
```diff
@@ -139,11 +161,15 @@ function getPackager () {
...
+  if (pnpm.isInstalled()) {
+    return pnpm
+  }
+
   if (npm.isInstalled()) {
     return npm
   }
 
-  warn('Please install Yarn or NPM before running this command.\n')
+  warn('Please install Yarn or PNPM or NPM before running this command.\n')
```
Comment (yusufkandemir) [inline app-webpack/lib/helpers/node-packager.js]:
```suggestion
  warn('Please install Yarn, PNPM, or NPM before running this command.\n')
```

**Comment by yusufkandemir** [comment]
Thanks for the contribution 👏 This will be included in the next releases of app-vite and app-webpack. Documentation for PNPM, create-quasar updates, etc. will be followed later.

**Pull request merged**